### PR TITLE
#15051. Implement new chatd commands MSGIDTIMESTAMP and NEWMSGIDTIMESTAMP

### PR DESCRIPTION
**Pull request by alber2510**

### CODE REVIEW

File: src/chatd.cpp [hunk 1]
```diff
@@ -3815,7 +3833,15 @@ Idx Chat::msgConfirm(Id msgxid, Id msgid)
         CHATD_LOG_DEBUG("msgConfirm: updated opcode MSGUPDx to MSGUPD and the msgxid=%u to msgid=%u of %d message/s in the sending queue", msgxid, msgid, count);
     }
 
-    CALL_LISTENER(onMessageConfirmed, msgxid, *msg, idx);
+    bool tsUpdated = false;
+    if (timestamp != 0)
+    {
+        msg->ts = timestamp;
```
Comment (sergiohs84) [inline src/chatd.cpp:3839]: The `ts` is updated in RAM, but not in DB

File: src/chatd.h [hunk 2]
```diff
@@ -1426,7 +1427,9 @@ class Client : public karere::DeleteTrackable
     //  * Changes at CALLDATA protocol (new state)
     // - Version 6:
     //  * Add commands ADDREACTION DELREACTION REACTIONSN
-    static const unsigned chatdVersion = 6;
+    // - Version 7:
+    //  * Add commands MSGIDTIMESTAMP OP_NEWMSGIDTIMESTAMP
```
Comment (sergiohs84) [inline src/chatd.h:1431]:
```suggestion
    //  * Add commands MSGIDTIMESTAMP NEWMSGIDTIMESTAMP
```

File: src/chatdMsg.h [hunk 3]
```diff
@@ -431,7 +431,11 @@ enum Opcode
       */
     OP_REACTIONSN = 48,
 
-    OP_LAST = OP_HANDLELEAVE,
+    OP_MSGIDTIMESTAMP = 49,
```
Comment (sergiohs84) [inline src/chatdMsg.h:434]: Missing documentation for those 2 new opcodes. It should indicate the size of params, as usual, and the purpose.

File: src/chatd.cpp (hunk 1 above)
Comment (alber2510) [reply src/chatd.cpp]: Solved it

File: src/chatdMsg.h (hunk 3 above)
Comment (alber2510) [reply src/chatdMsg.h:442]: Solved it

**Pull request merged**